src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -6467,11 +6468,21 @@ def create_vnet(cmd, resource_group_name, vnet_name, vnet_prefixes='10.0.0.0/16'
     if bgp_community is not None and cmd.supported_api_version(min_api='2020-06-01'):
         VirtualNetworkBgpCommunities = cmd.get_models('VirtualNetworkBgpCommunities')
         vnet.bgp_communities = VirtualNetworkBgpCommunities(virtual_network_community=bgp_community)
+    if enable_encryption is not None:
+        if not vnet.encryption:
+            vnet.encryption = {}
```

Comment:
with this block, then line 6476 would never be hit?

Reply:
The line 6474 will ensure that vnet.encryption = {"enabled": true|false}.
And vnet.encryption can be none sometimes, 'enabled' in vnet.encryption will raise TypeError: argument of type 'NoneType' is not iterable.